### ISSUE TITLE
Is it time to use PowerShell to configure IIS

### ISSUE THREAD

**Issue by tas50** [chef-cookbooks/iis#451]
Now that Windows 2008 R2 is EOL is it fair to say we can use PowerShell to configure IIS? It seems a lot of complexity and issues in this cookbook could be resolved by using PowerShell cmdlets.

**Comment by EasyAsABC123** [comment]
Yes 100% I’ve wanted to rewrite this using the DSC to solve most of these
problems but the decision sees to never be finalized on chefs side

**Comment by jakauppila** [comment]
Using DSC would require pulling in additional dependencies though, right?

I had started looking at what a conversion would look like with #424 a few years ago, but hadn't had the time to commit to seeing it through.

**Comment by EasyAsABC123** [comment]
Yes you need to download the DSC powershell and then install it, that isn't a huge deal.  It depends on pinning the version or not, in my mind you'd download the latest at the start of every run.  it means it would break for everyone but you could enable pinning via a configuration setting you pass to iis.

I looked into this a while back and had figured all that out, but they couldn't ever get their minds set...since then i've become more aware of the lack of support Chef can provide for legacy config items.  I say just implement it.

**Comment by jakauppila** [comment]
I would want to be mindful of air-gapped environments to ensure that the installation of the modules is easy to configure from alternative sources.

That being said, I'm not sure how often the WebAdministration DSC module updates, but I'd personally be more in favor of a pure lower-level WebAdministration module configuration that does not rely on external dependencies since you would then have to manage that as well in case there are bugs and/or breaking changes in the DSC modules versions.

**Comment by Xorima** [comment]
When I used DSC resources for other chef community resources I have had negative feedback on that aspect, I think using raw powershell would be better
